### PR TITLE
Fallback to pillow

### PR DESCRIPTION
**Pull request by Hakuyume**
I modify `resize` function to use `Pillow` if `cv2` is not installed.

### CODE REVIEW

File: chainercv/transforms/image/resize.py [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+    warnings.warn(
+        'cv2 is not installed on your environment. '
+        'ChainerCV will fallback on Pillow. ',
+        RuntimeWarning)
+
+    def _resize(img, size):
+        channels = []
+        for i in range(3):
+            ch = PIL.Image.fromarray(img[:, :, i], mode='F')
+            ch = ch.resize(size)
```
Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Could you add a keyword argument `resample=PIL.Image.BILINEAR`?

File: chainercv/transforms/image/resize.py [hunk 2]
```diff
@@ -21,15 +39,8 @@ def resize(img, output_shape):
         ~numpy.ndarray: A resize array in CHW format.
```
Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Please add a documentation about modules used depending on the environment (CV2, PIL).

Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Also, please add that cv2 is the faster implementation.

File: chainercv/transforms/image/resize.py [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+        return cv2.resize(img, dsize=size)
 
 except ImportError:
-    _available = False
+    import numpy
+    import PIL
+
+    warnings.warn(
+        'cv2 is not installed on your environment. '
+        'ChainerCV will fallback on Pillow. ',
```
Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Please note that CV2 is faster

**Comment by yuyu2172** [comment]
Thanks.
I benchmarked both implementation and concluded that PIL is slower than cv2.
Do you agree with this?

The code I used for benchmarking is below.

```python
from skimage.data import astronaut
import numpy as np
import time

import cv2
import PIL


def _resize_cv2(img, size):
    return cv2.resize(img, dsize=size)


def _resize_pillow(img, size):
    channels = []
    for i in range(3):
        a = img[:, :, i]
        ch = PIL.Image.fromarray(img[:, :, i], mode='F')
        ch = ch.resize(size)
        channels.append(np.array(ch))
    return np.stack(channels, axis=2)

size = (256, 128)
img = astronaut().astype(np.float32)

for i in range(5):
    _resize_pillow(img, size)

times = []
for i in range(10):
    start = time.time()
    _resize_pillow(img, size)
    times.append(time.time() - start)
print('pillow ', np.mean(times))


for i in range(5):
    _resize_cv2(img, size)

times = []
for i in range(10):
    start = time.time()
    _resize_cv2(img, size)
    times.append(time.time() - start)
print('cv2 ', np.mean(times))

```

File: chainercv/transforms/image/resize.py (hunk 2 above)
Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Also, please add that `img` should be float type.

**Comment by Hakuyume** [comment]
Thank you for your benchmark code.
On my environment, this code returned as follows.
```
pillow  0.00493938922882
cv2  0.00032696723938
```
So, we can say that `cv2` is much faster than `Pillow`.

**Comment by yuyu2172** [comment]
LGTM!

**Pull request merged**